### PR TITLE
Changes to prepare the integration of the Tracker upgrade geometries

### PR DESCRIPTION
**Pull request by venturia**

### CODE REVIEW

**Comment by venturia** [comment]
@boudoul can be interested.

This pull request comes with (at least) four caveats:
1) phase 1 and phase 2 workflows should run, including offline track reconstruction BUT there is a (non-fatal) error in the TrackAssociator because the link between TOB subdetector and SiStripClusters is no more there. A different way to decide which cluster collection has to be looked at, has to be implemented using the updated "isPixel" "isStrip" methods. 
2) The Tracker Trigger code likely will not work and this pull request has to be considered as an opportunity to define the work to do.
3) Some changes have been applied to the XML geometry files but the same changes have NOT been propagated to the same files in the "dev" areas introduced since SLHC20. It should be simple to identify what are the changes to be applied by comparing this pull request to what was in SLHC20
4) Even if the local and track reconstructions work nicely in this release, this does not mean that it will be easy for 75x because in this release we are still profiting of the hardcoded detid classes which are configured for the phase 1 pixel detector by default.

**Comment by mark-grimes** [comment]
There are some errors in unit tests, I'm guessing the tests probably have to be rewritten for the new geometry methods.  Not overly worried about those (for now).

This breaks the 4 eta SHCal and track trigger tests (126xx and 128xx respectively).  The SHCal one looks like a minor typo somewhere in the geometry ([link to full log](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-8670/4117/runTheMatrix-results/12600_FourMuPt1_200+FourMuPt_1_200_Extended2023SHCal4Eta_GenSimHLBeamSpotFull+DigiFull_Extended2023SHCal4Eta+RecoFull_Extended2023SHCal4Eta+HARVESTFull_Extended2023SHCal4Eta/step1_FourMuPt1_200+FourMuPt_1_200_Extended2023SHCal4Eta_GenSimHLBeamSpotFull+DigiFull_Extended2023SHCal4Eta+RecoFull_Extended2023SHCal4Eta+HARVESTFull_Extended2023SHCal4Eta.log)):

    ImportError: No module named trackerNumberin2023Geometry_cfi

Think I've found it and can patch myself.

The track trigger one ([link to full log](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-8670/4117/runTheMatrix-results/12800_FourMuPt1_200+FourMuPt_1_200_Extended2023TTI_GenSimHLBeamSpotFull+DigiTrkTrigFull_Extended2023TTI/step2_FourMuPt1_200+FourMuPt_1_200_Extended2023TTI_GenSimHLBeamSpotFull+DigiTrkTrigFull_Extended2023TTI.log)) looks like a more serious geometry problem.  No idea about that one.

Since this is a development branch this doesn't necessarily preclude it from being merged, although this has to be on the understanding that the track trigger problem is investigated and fixed, or that a track trigger rewrite/refactoring further down the road will solve the problem.

File: Configuration/Geometry/python/GeometryExtended2023HGCal_cff.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import FWCore.ParameterSet.Config as cms
 
 from Geometry.CMSCommonData.cmsExtendedGeometry2023HGCalXML_cfi import *
-from Geometry.TrackerNumberingBuilder.trackerNumberingSLHCGeometry_cfi import *
+from Geometry.TrackerNumberingBuilder.trackerNumberin2023Geometry_cfi import *
```
Comment (mark-grimes) [inline Configuration/Geometry/python/GeometryExtended2023HGCal_cff.py]: Guess you're missing a "g" here?  Can you change or do you want me to?

File: Configuration/Geometry/python/GeometryExtended2023SHCal4Eta_cff.py [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import FWCore.ParameterSet.Config as cms
 
 from Geometry.CMSCommonData.cmsExtendedGeometry2023SHCal4EtaXML_cfi import *
-from Geometry.TrackerNumberingBuilder.trackerNumberingSLHCGeometry_cfi import *
+from Geometry.TrackerNumberingBuilder.trackerNumberin2023Geometry_cfi import *
```
Comment (mark-grimes) [inline Configuration/Geometry/python/GeometryExtended2023SHCal4Eta_cff.py]: And a "g" here too.

**Comment by venturia** [comment]
I have fixed the typos
For the Track Trigger: breaking it was expected. That part has to be reviewed and made compatible to the new design (and/or the new design has to be modified to accomodate more easily the Track Trigger part)

**Comment by mark-grimes** [comment]
merge

Okay.  We'll put it in the IB and worry about the track trigger later.

**Comment by boudoul** [comment]
I'm preparing a PR with the dev xml fixed as mentioned in Point 3)

**Pull request merged**